core/src/main/java/io/grpc/CallCredentials.java

```diff
@@ -65,6 +72,7 @@
    * @param applier The outlet of the produced headers. It can be called either before or after this
    *        method returns.
    */
+  @ExperimentalApi("https//github.com/grpc/grpc-java/issues/1914")
   void applyRequestMetadata(
```

Comment:
They can implement it, but it needs to be a base class.  Experimental API and interface are all or nothing.  

> I don't understand that. I'd have understood "base class than this extends" and "make this a base class that is extended".

Something like: 
```
interface CallCredentials {
  void doNotImplementThisOrYouWillBreak();
}
public abstract class BaseCallCredentials implements CallCredentials {
public final void doNotImplementThisOrYouWillBreak();
} 
```

That way CallCredentials can mostly live on as is, Experimental methods can be moved down to the base class, where they can change.  If methods change, callers of experimental methods will change, but implementors of CallCredentials will not.

As the code stands with this PR, someone extending CallCredentials, a stable class, will suffer breakage when experimental methods come and go.  That is not stable.